elastic/fleet-server
# Separate logger level reload from config reload

## What is the problem this PR solves?

Subsystems, such as the ES bulker and API listeners restart whenever the log level is changed.

## How does this PR solve the problem?

Split all logger config changes from server config changes so that a change to the fleet-server logging level will not force the api server or other subsystens to restart. Use zerolog.SetGlobalLevel to control log level instead of creating a new global logger when the log level changes.

## How to test this PR locally

Enroll a fleet-server instance that runs under the elastic-agent.
Change the log level through the Kibana UI.
The server restart message:
```
{"log.level":"info","@timestamp":"2023-03-24T20:26:57.639Z","message":"starting server on configuration change","component":{"binary":"fleet-server","dataset":"elastic_agent.fleet_server","id":"fleet-server-default","type":"fleet-server"},"log":{"source":"fleet-server-default"},"service.name":"fleet-server","ecs.version":"1.6.0","ecs.version":"1.6.0"}
```
should not appear as a part of the fleet-server logs.

## Checklist

- ~~I have made corresponding changes to the documentation~~
- ~~I have made corresponding change to the default configuration files~~

## Related issues

- Closes #2453
